File: src/main/kotlin/com/github/magneticflux/aoc/day2/Problem1.kt
package com.github.magneticflux.aoc.day2

import com.github.magneticflux.aoc.input
import java.util.*
import kotlin.math.absoluteValue

/**
 * Created by Mitchell Skaggs on 12/1/2017.
 */
object Problem1 {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(input())
        input.useDelimiter("\n")
        val grid = input.asSequence().map { it.split('\t').map { it.toInt() }.sorted() }.toList()
        println(grid.map { (it.first() - it.last()).absoluteValue }.sum())
    }
}

File: src/main/scala/com/github/magneticflux/aoc/year2018/day1/Problem1.scala
package com.github.magneticflux.aoc.year2018.day1

import com.github.magneticflux.aoc.InputHelpers._

/**
  * Created by Mitchell Skaggs on 12/2/2018.
  */
object Problem1 {
  def main(args: Array[String]): Unit = {
    val answer = this.input()
      .getLines()
      .map(_.toInt)
      .sum

    println(answer)
  }
}

